### ISSUE TITLE
Is multi-threading supported?

### ISSUE THREAD

**Issue by GRASBOCK** [nestordemeure/Simplers#5]
I like that it's a black box. 
Does this crate support multi-threading though? I tried but I only get about 40% CPU utilization.

**Comment by lovasoa** [comment]
@nestordemeure : Would you be interested in a pull request that implements multi-threading ? Using rayon, for instance.

**Comment by nestordemeure** [comment]
Yes @Iovasoa, I would hapilly accept a pull request introducing multithreading :)

It should not be too hard to do, the main constraint would be to make it optional (paralelism is not desirable when the evaluation function is already parallel), if possible with a feature (to avoid compiling rayon when one does not need it).

@GRASBOCK sorry for the late reply, I am only seeing your issue now. The code is not parrallel at the moment.

**Comment by nestordemeure** [comment]
Closing this issue as it has been inactive. Plus, the algorithm is inherently  sequential, most performance gains would come from work on the evaluation function.
